server/app.py

```diff
@@ -88,18 +92,36 @@ async def _calculate_fps_loop(self):
                 current_time = time.monotonic()
                 if self._last_fps_calculation_time is not None:
                     time_diff = current_time - self._last_fps_calculation_time
-                    self._fps = self._fps_interval_frame_count / time_diff
+                    self._fps = (
+                        self._fps_interval_frame_count / time_diff
+                        if time_diff > 0
+                        else 0.0
+                    )
                     self._fps_measurements.append(
                         {
                             "timestamp": current_time - self._fps_loop_start_time,
                             "fps": self._fps,
                         }
                     )  # Store the FPS measurement with timestamp
 
-                # Reset start_time and frame_count for the next interval.
+                    # Store the average FPS over the last minute.
```

Comment:
Moved here to prevent redundant calculations and makes it available for prometheus serving.